Software/bsw/static/Mcal/CAN/src/Can.c

```diff
@@ -1689,51 +1720,49 @@ return  Return_type;
 Std_ReturnType Can_GetControllerMode(uint8 Controller,Can_ControllerStateType* ControllerModePtr) 
 {
 	Std_ReturnType Loc_Can_GetControllerMode_Ret = E_OK;
-
-#if(CanDevErrorDetect == STD_ON)
-	if (CAN_CS_UNINIT==ControllerState[Controller])
-	{
-		Det_ReportError(CAN_MODULE_ID,CAN_INSTANCE_ID,Can_GetControllerMode_Id,CAN_E_UNINIT);
-		Loc_Can_GetControllerMode_Ret = E_NOT_OK;
-	}
-	else
-	{
-	    /* Do Nothing */
-	}
-
-#endif
-
-	if ( NUM_OF_CAN_CONTROLLERS <= Controller )
-	{
-		Det_ReportError(CAN_MODULE_ID, CAN_INSTANCE_ID,Can_GetControllerMode_Id, CAN_E_PARAM_CONTROLLER);
-		Loc_Can_GetControllerMode_Ret = E_NOT_OK;
-	} 
-	else 
+    /*  [SWS_Can_91016] If development error detection for the Can module is enabled:
+     *  The function Can_GetControllerMode shall raise the error CAN_E_UNINIT and
+     *  return E_NOT_OK if the driver is not yet initialized.
+     *  ( SRS_BSW_00406,SRS_BSW_00416)
+     */
+	if(ModuleState == CAN_UNINIT)
 	{
-	    /* Do Nothing */
+        #if(CanDevErrorDetect == STD_ON)
```

Comment:
I'd prefere a single #if(CanDevErrorDetect == STD_ON) over the consecutive checks.

Reply:
Okay . I will modify it .

Reply:
No, Actually I made it like that because the function can't return the Controller_mode
if any error occured . So these if conditions must be part of the code if the DET is disabled 
![image](https://user-images.githubusercontent.com/32249454/70515476-0f7d1780-1b3e-11ea-91b6-e715a58d514e.png)

Reply:
If so, the  Loc_Can_GetControllerMode_Ret = E_NOT_OK; should be after the #endif, right?